### PR TITLE
Blog: pgvector fewer dimensions post

### PR DESCRIPTION
**Pull request by gregnr**
Adds "pgvector: Fewer dimensions are better" blog post.

### Preview
https://zone-www-dot-com-git-feat-fewer-dimensions-blog-post-supabase.vercel.app/blog/fewer-dimensions-are-better-pgvector

### Todo
- [x] create thumbnail images
- [x] add inline images

### CODE REVIEW

File: apps/www/_blog/2023-08-03-fewer-dimensions-are-better-pgvector.mdx [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+  />
+</div>
+
+Dot product is the product of each vector element pair summed together into a single result. Fewer dimensions in the vector means fewer calculations for every computed distance .
+
+We've compared `text-embedding-ada-002` from OpenAI (1536 dimensions) with open-source `all-MiniLM-L6-v2` (384 dimensions), using the following configuration:
+
+- **Database size:** with 2vCPU (ARM) and 8GB RAM - `large` add-on for Supabase project.
+- **Version:** Postgres v15 and pgvector v0.4.0
+- **Dataset:** 224,482 embeddings from Wikipedia articles.
```
Comment (gregnr) [inline apps/www/_blog/2023-08-03-fewer-dimensions-are-better-pgvector.mdx:92]: @olirice do we have a link to the wikipedia article dataset?

File: apps/www/_blog/2023-08-03-fewer-dimensions-are-better-pgvector.mdx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+- Cosine distance
+- Euclidean distance
+
+## Challenges with pgvector
+
+Without indexes, pgvector performs a full table scan when you run a similarity query. This means distance has to be computed against every row in your table. This is manageable at a small scale, but becomes problematic as your table grows.
+
+To solve this, pgvector offers indexes. Indexes reorganize the data into data structures that exploit internal structure and enable approximate similarity search without referring to every record. Currently, pgvector supports an IVF index, with HNSW expected in the next release.
+
+IVF indexes work by clustering vectors into `lists`, and then querying only vectors within the same list (or multiple nearby lists, depending on the value of `probes`). To learn more about pgvector's IVF indexes and its different configurations, read the docs on [Managing indexes](https://supabase.com/docs/guides/ai/managing-indexes).
```
Comment (kiwicopple) [inline apps/www/_blog/2023-08-03-fewer-dimensions-are-better-pgvector.mdx:30]: > To learn more about pgvector's IVF indexes and its different configurations, read the docs on Managing indexes

I'd probably remove this (it breaks the flow). If you want to link to the page make it more "inline" - eg:

```
IVF [indexes](https://supabase.com/docs/guides/ai/managing-indexes) work by clustering vectors into `lists`, and then querying only vectors within the same list (or multiple nearby lists, depending on the value of `probes`).
```

File: apps/www/_blog/2023-08-03-fewer-dimensions-are-better-pgvector.mdx [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+- Classification
+- Clustering
+- Pair classification
+- Reranking
+- Summarization
+- Bitext mining
+
+Models are ranked per task by taking their average score produced over each task's datasets. Each model also has an overall (general purpose) score, calculated by taking the average score produced across all datasets. You can find the results on their [MTEB leaderboard](https://huggingface.co/spaces/mteb/leaderboard).
+
+Looking at these results, it's worth pointing out that each model's dimension size has little-to-no correlation with its performance. In fact there are a number of models that perform comparably with`text-embedding-ada-002`, all of which produce embeddings with fewer dimensions than 1536.
```
Comment (kiwicopple) [inline apps/www/_blog/2023-08-03-fewer-dimensions-are-better-pgvector.mdx:55]:
```suggestion
It's worth pointing out that each model's dimension size has little-to-no correlation with its performance. In fact there are a number of models that perform comparably with`text-embedding-ada-002`, all of which produce embeddings with fewer dimensions than 1536.
```

**Review by kiwicopple — APPROVED** [review]

File: apps/www/_blog/2023-08-03-fewer-dimensions-are-better-pgvector.mdx (hunk 1 above)
Comment (gregnr) [reply apps/www/_blog/2023-08-03-fewer-dimensions-are-better-pgvector.mdx:92]:
```suggestion
- **Dataset:** 224,482 embeddings from [Wikipedia articles](https://www.kaggle.com/datasets/stephanst/wikipedia-simple-openai-embeddings).
```

**Pull request merged**